arangod/GeneralServer/AcceptorTcp.cpp

```diff
@@ -189,17 +186,15 @@ void AcceptorTcp<SocketType::Ssl>::performHandshake(std::unique_ptr<AsioSocket<S
     if (ec) {  // canceled
       return;
     }
-    asio_ns::error_code err;
-    ptr->shutdown(err);  // ignore error
+    ptr->shutdown([](asio_ns::error_code const&) {});  // ignore error
   });
 
   auto cb = [this, as = std::move(proto)](asio_ns::error_code const& ec) mutable {
     as->timer.cancel();
     if (ec) {
       LOG_TOPIC("4c6b4", DEBUG, arangodb::Logger::COMMUNICATION)
           << "error during TLS handshake: '" << ec.message() << "'";
-      asio_ns::error_code err;
-      as->shutdown(err);  // ignore error
+      as.reset(); // ungraceful shutdown
```

Comment:
https://www.boost.org/doc/libs/1_71_0/doc/html/boost_asio/reference/basic_stream_socket/close/overload2.html -- see Remarks.

Why is it ok to kill / reset the connection? Will there be no problem for clients (They could wait for a response on a connection that is not closed for them)?